iternity-dotcom/minio
# WIP: Implement distributed locking with etcd

## Description

Implement distributed locking with etcd to allow for better performance using `backgroundAppend()` in a load balanced cluster scenario. We can of course only benefit from this when using a setup with etcd, but that is also our default use case.

## Motivation and Context

We've observed that, in a distributed setup with a load balancer, completing a multipart upload takes a long time. The reason for this is because the existing `backgroundAppend()` algorithm is not cluster aware.

This leads to #11 and #13 .

## How to test this PR?

- A manual test can easily verify if the CompleteMultipartUpload behavior has improved.
- Running unit tests should ensure that we didn't break anything.
- Running the Mint test suite will give us even more confidence.

## Types of changes

## Checklist:
